### ISSUE TITLE
Groups & deleting

### ISSUE THREAD

**Issue by pribism** [oskopek/StudyGuide#25]
It would be good if user could manually specify the number or name of the group, subject falls in and he would see how many ECTS he has already achieved/enrolled in that group. For our school one of the most important things. 

Also in the GUI it would be great if you could also delete subject from your internal database, I've imported it more times and I have it in the menu add course repeated few times.

**Comment by oskopek** [comment]
Won't get around to fixing this any time soon. You're welcome to bug me to fix this sooner, maybe I will afterwards :smile: 

**Temporary workaround:**

* Open the saved file in a text editor
* Go to `"constraints"`
* Go to `"courseGroupConstraintList"`
* Add an entry like so:

1. Choose if you want `CourseGroupCreditsSumConstraint` or `CourseGroupFulfilledAllConstraint`
  * Lets assume `CourseGroupCreditsSumConstraint`. The other is easy, just omit `totalNeeded`
2. Add this into `"courseGroupConstraintList" : [HERE]`:
```json
    {
      "CourseGroupCreditsSumConstraint" : {
        "courseGroup" : {
          "courseList" : [LIST_OF_COURSE_IDS]
        },
        "totalNeeded" : {
          "@id" : "make-up-a-random-hex-string-here",
          "creditValue" : YOUR-TARGET-CREDIT-NUMBER
        }
      }
    }
```
  * For `LIST_OF_COURSE_IDS`, just copy and paste course `"@id"`s (**NOT** `"id"`s) entries from the `"courseRegistry"`, separated by commas. Complete example:


```json
{
      "CourseGroupCreditsSumConstraint" : {
        "courseGroup" : {
          "courseList" : [ "3ff30633-9b59-40e8-aabf-9370734560a9", "3d3d7126-49e0-4211-a209-29a8639a061c", "9d835d98-3eaf-42d0-8c61-476ced71aa87" ]
        },
        "totalNeeded" : {
          "@id" : "e3c18828-c131-4115-87fa-6710d96feee7",
          "creditValue" : 6
        }
      }
    }
```

Hope this works, if not, let me know.

**Comment by pribism** [comment]
Yes, that raises a broken group constraint, doesn't notify me which however, nor notifies me how to fix it, with 5 groups as we have it would soon become intricate.

Would be nice (for the future, it isn't a necessity, rather an enhancement) if the user could specify/make groups and assign subjects to the groups in the GUI and the system would show how many ECTS he has from that group and how many is he missing. But it's more of an icing on a cake ;)

**Comment by oskopek** [comment]
Thanks for the feedback :smile:  Will keep this open until I get around to fixing it.
